validator/src/main/java/io/avaje/validation/spi/AdapterFactory.java

```diff
@@ -0,0 +1,21 @@
+package io.avaje.validation.spi;
+
+import java.lang.reflect.Type;
+
+import io.avaje.validation.adapter.ValidationAdapter;
+import io.avaje.validation.adapter.ValidationContext;
+
+/** Factory for creating a ValidationAdapter for a given type. */
+@FunctionalInterface
+public non-sealed interface AdapterFactory extends ValidationExtension {
```

Comment:
Note to self, check language level - non-sealed.

Reply:
?

Reply:
It's a note for me to check/reminder on why we made this JDK 17+ versus leaving it at JDK 11. I can't remember why we did that.

Reply:
Nice :). Ta !!